src/Std/Data/HashMap/Lemmas.lean

```diff
@@ -1269,4 +1269,330 @@ theorem isEmpty_unitOfList [EquivBEq α] [LawfulHashable α]
+        get m k' h' :=
+  DHashMap.Const.get_alter
+
+@[simp]
+theorem get_alter_self [EquivBEq α] [LawfulHashable α] {k : α} {f : Option β → Option β}
+    {h : k ∈ alter m k f} :
+    haveI h' : (f (get? m k)).isSome := mem_alter_self |>.mp h
+    get (alter m k f) k h = (f (get? m k)).get h' :=
+  DHashMap.Const.get_alter_self
+
+theorem get!_alter [EquivBEq α] [LawfulHashable α] {k k' : α} [Inhabited β]
+    {f : Option β → Option β} : get! (alter m k f) k' =
+      if k == k' then
+        f (get? m k) |>.get!
+      else
+        get! m k' :=
+  DHashMap.Const.get!_alter
+
+@[simp]
+theorem get!_alter_self [EquivBEq α] [LawfulHashable α] {k : α} [Inhabited β]
+    {f : Option β → Option β} : get! (alter m k f) k = (f (get? m k)).get! :=
+  DHashMap.Const.get!_alter_self
+
+theorem getD_alter [EquivBEq α] [LawfulHashable α] {k k' : α} {v : β} {f : Option β → Option β} :
+    getD (alter m k f) k' v =
+      if k == k' then
+        f (get? m k) |>.getD v
+      else
+        getD m k' v :=
+  DHashMap.Const.getD_alter
+
+@[simp]
+theorem getD_alter_self [EquivBEq α] [LawfulHashable α] {k : α} {v : β} {f : Option β → Option β} :
+    getD (alter m k f) k v = (f (get? m k)).getD v :=
+  DHashMap.Const.getD_alter_self
+
+theorem getKey?_alter [EquivBEq α] [LawfulHashable α] {k k' : α} {f : Option β → Option β} :
+    (alter m k f).getKey? k' =
+      if k == k' then
+        if (f (get? m k)).isSome then some k else none
+      else
+        m.getKey? k' :=
+  DHashMap.Const.getKey?_alter
+
+theorem getKey?_alter_self [EquivBEq α] [LawfulHashable α] {k : α} {f : Option β → Option β} :
+    (alter m k f).getKey? k = if (f (get? m k)).isSome then some k else none :=
+  DHashMap.Const.getKey?_alter_self
+
+theorem getKey!_alter [EquivBEq α] [LawfulHashable α] [Inhabited α] {k k' : α}
+    {f : Option β → Option β} : (alter m k f).getKey! k' =
+      if k == k' then
+        if (f (get? m k)).isSome then k else panic ""
+      else
+        m.getKey! k' :=
+  DHashMap.Const.getKey!_alter
+
+theorem getKey!_alter_self [EquivBEq α] [LawfulHashable α] [Inhabited α] {k : α}
+    {f : Option β → Option β} :
+    (alter m k f).getKey! k = if (f (get? m k)).isSome then k else panic "" :=
+  DHashMap.Const.getKey!_alter_self
+
+theorem getKey_alter [EquivBEq α] [LawfulHashable α] [Inhabited α] {k k' : α}
+    {f : Option β → Option β} {h : k' ∈ alter m k f} :
+    (alter m k f).getKey k' h =
+      if heq : k == k' then
+        k
+      else
+        haveI h' : k' ∈ m := mem_alter_of_beq_eq_false (Bool.not_eq_true _ ▸ heq) |>.mp h
+        m.getKey k' h' :=
+  DHashMap.Const.getKey_alter
+
+@[simp]
+theorem getKey_alter_self [EquivBEq α] [LawfulHashable α] [Inhabited α] {k : α}
+    {f : Option β → Option β} {h : k ∈ alter m k f} :
+    (alter m k f).getKey k h = k :=
+  DHashMap.Const.getKey_alter_self
+
+theorem getKeyD_alter [EquivBEq α] [LawfulHashable α] {k k' d : α} {f : Option β → Option β} :
+    (alter m k f).getKeyD k' d =
```

Comment:
I think so far we have been using `fallback` as the name for the fallback in theorem statements.